src/dmd/root/filename.d

```diff
@@ -64,11 +64,28 @@ nothrow:
         return compare(str, (cast(FileName*)obj).str);
     }
 
+    private static char normalizeWinPathChar(char c) pure
+    {
+        if (c >= 'a' && c <= 'z') return cast(char)(c - ('a' - 'A'));
```

Comment:
Uppercase file name characters are annoying, better to normalize to lower case. Also, this should be a nested function within `compare()`.

Reply:
The character being returned isn't used to build a new filename string, it's just used as a temporary value for comparision.

From what I understand, comparison functions usually convert to upper case because it will typically result in less if statements.  Because the lower-case letters are at the end of the ASCII table, most characters will be less than `'a'` which means they will only perform one comparison.  If we go to lower-case instead, then you'll get have
```D
if (c >= 'A' && c <= 'Z') ...
```
then you'll almost always perform 2 comparisons whether or not c is upper case or lower case, resulting in more total comparisons.